src/cwhy/cwhy.py

```diff
@@ -195,38 +215,20 @@ def __init__(self, args, diagnostic):
         max_code_locations = args["max_context"]
 
         # Go through the diagnostic and build up a list of code locations.
-        line = 0
-        while line < len(diagnostic_lines):
-            # This pattern works for some C++ compilers (GCC, Clang) and Rust.
-            match = re.match(
-                r"([a-zA-Z0-9./][^:->]+):([0-9]+):([0-9]+)", diagnostic_lines[line]
-            )
-
-            if not match:
-                # This pattern works for javac.
-                match = re.match(
-                    r"([a-zA-Z0-9./][^:->]+):([0-9]+):", diagnostic_lines[line]
-                )
-
-            if not match:
-                # This pattern works for Python, filtering out non-files (e.g., <string>).
-                match = re.match(
-                    r'\s*File "(.*?)", line (\d+), in ([^\<].*)', diagnostic_lines[line]
-                )
-
-            line += 1
-
-            if not match:
-                continue
+        for (linenum, line) in enumerate(diagnostic_lines):
+            for (lang, pattern) in error_patterns:
```

Comment:
Maybe we should detect if we match on conflicting languages, can be done in the future if the issue comes up.

Reply:
True. Let's see what Go errors look like :).

Reply:
Done, seems fine.

Reply:
Resolving this for now.